File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.java
class Solution {
    public int partitionString(String s) {
        int idx = 0;
        int count = 0;
        Map<Character, Boolean> mp = new HashMap<>(); 
        while (idx < s.length()) {
            if (mp.containsKey(s.charAt(idx))) {
                count++; 
                mp.clear(); 
            }
            mp.put(s.charAt(idx), true); 
            idx++;
        }
        return ++count;
    }
}

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution(object):
    def partitionString(self, s):
        idx = 0
        count = 0
        mp = {}
        while idx < len(s):
            if s[idx] in mp: 
                count += 1
                mp.clear()
            mp[s[idx]] = True
            idx += 1
        return count + 1